cmd/rook/ceph/cleanup.go

```diff
@@ -58,21 +62,63 @@ func startCleanUp(cmd *cobra.Command, args []string) error {
 			logger.Infof("successfully cleaned up %q directory", cleanupDirPath)
 		}
 		// Remove all the mon directories.
-		monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
+		if monSecret == "" {
+			return errors.Errorf("failed to clean up mon directories in the dataDirHostPath %q due to empty monitor secret", dataDirHostPath)
```

Comment:
We wouldn't expect this condition, right? Seems like the operator is always guaranteeing it will pass the key anyway. If it was a possible condition I would suggest to just log the error and skip `cleanMonDir()`, but allow the cleanup to continue. This would be more important when device cleanup is added.